File: Test.Web/src/app/pages/models/meteorite-context.ts
export interface MeteoritesContext {
  readonly data: readonly MeteoriteData[];
  readonly possibleYears: readonly number[];
  readonly possibleClasses: readonly string[];
}

export interface MeteoriteData {
  readonly id: string;
  readonly name: string;
  readonly class: string;
  readonly recclass: string;
  readonly mass: number;
  readonly fall: string;
  readonly year: number;
  readonly reclat: number;
  readonly reclong: number;
}

File: Test.Core/Models/MeteoriteContext.cs
using Test.DataAccess.Entities;

namespace Test.Core.Models;

public class MeteoriteContext
{
    public List<MeteoriteData> Data { get; set; }
    public List<int> PossibleYears { get; set; }
    public List<string> PossibleClasses { get; set; }
}

public class MeteoriteData
{
    public Guid Id { get; init; }
    public string Name { get; init; }
    public string Recclass { get; init; }
    public float Mass { get; init; }
    public string Fall { get; init; }
    public int Year { get; init; }
    public float Reclat { get; init; }
    public float Reclong { get; init; }

    public MeteoriteData(Meteorite entity)
    {
        ArgumentNullException.ThrowIfNull(entity);

        Id = entity.Id;
        Name = entity.Name;
        Recclass = entity.Class.Name;
        Mass = entity.Mass;
        Fall = entity.Fall.ToString();
        Year = entity.Year.Year;
        Reclat = entity.Reclat;
        Reclong = entity.Reclong;
    }
}
